Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +124,155 @@ public void Create_PropertyDeclaredInBaseClass ()
+
+      var notNullValidators1 = ((IPropertyValidationRule)result1).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators1.Length, Is.EqualTo(1));
+      Assert.That(notNullValidators1[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result2 = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock2.Object);
+
+      var notNullValidators2 = ((IPropertyValidationRule)result2).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators2.Length, Is.EqualTo(2));
+      Assert.That(notNullValidators2[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
+      Assert.That(notNullValidators2[1].ValidationMessage.ToString(), Is.EqualTo("newTest"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_IgnoresValidationMessagesForValidatorsWithoutADeferredInitializationValidationMessage ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      var propertyValidators = ((IPropertyValidationRule)result).Validators;
+
+      Assert.That(propertyValidators, Is.Empty);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WithValidationMessageFactoryReturnsNull_ThrowsInvalidOperationException ()
     {
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns((ValidationMessage)null);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+
+      Assert.That(
+          () => _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object),
+          Throws.InvalidOperationException
+              .With.Message.EqualTo("The IValidationMessageFactory did not return a result for NotNullValidator applied to property 'UserName' on "
+                                    + "type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WithConditionNotNull_InitializesConditionForCreatedPropertyValidationRule ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
```

Comment:
We also have the InvariantValidationMessage. My personal style is using that one because it's more straight forward. In the end, it doesn't really matter which version you use, it's just a case of simplifying the test and best practice / a typical pattern. In more advanced scenarios such simplfiications could become a code-style requirement instead of good practice in order to keep things readable.